### ISSUE TITLE
Var conflict when dom-repeat has as="document"

### ISSUE THREAD

**Issue by eheikes** [Banno/polymer-rename#7]
If there is a `dom-repeat` that sets the local loop variable to `document`, e.g.

```html
<template is="dom-repeat" items="[[documents]]" as="document">
  <jha-link-button icon href="#"></jha-link-button>
</template>
```

then the generated code will try to reference that local var when using `document.createElement` to create sub-elements:

```javascript
  for (let index = 0; index < this.documents.length; index++) {
    let document = this.documents[index];
    // ...
    {
      let polymerRename_jha_link_buttonElement = /** @type {!JhaLinkButtonElement} */ (document.createElement('jha-link-button'));
    }
  }
```

which will of course create a warning in closure compiler:

```
WARNING - Property createElement never defined on document$100 of type module$src$js$api$models$document
      let polymerRename_jha_link_buttonElement = /** @type {!JhaLinkButtonElement} */ (document.createElement('jha-link-button'));
```

Can this be fixed, or if not, can we add a note to the docs?

**Comment by ChadKillingsworth** [comment]
Yowser. I can see if I can rework that. I'll have to think about how exactly ...

**Comment by ChadKillingsworth** [comment]
Closed by 99982a2d4b0a3c783a7ab267402905470003011a
